www/js/con-author.js

```diff
@@ -814,46 +818,63 @@ define([
-			this._model.student.setValue(newNodeID, currentNode.initial);
-			this._model.student.setUnits(newNodeID, currentNode.units);
-			
+			var description = this._model.authored.getDescription(currentNode.ID);
+			this._model.student.setDescription(newNodeID, description);
+			if(typeof currentNode.value !== 'undefined'){
+				this._model.student.setValue(newNodeID, currentNode.value);
+				this._model.student.setStatus(newNodeID, "value", {"disabled": true, "status": "correct"});
+			}
+			if(currentNode.units){
+				this._model.student.setUnits(newNodeID, currentNode.units);
+				this._model.student.setStatus(newNodeID, "units" , {"disabled":true,"status":"correct"});
+			}
+			if(currentNode.variableType){
+				this._model.student.setVariableType(newNodeID, currentNode.variableType);
+				this._model.student.setStatus(newNodeID, "variableType" , {"disabled":true,"status":"correct"});
+			}
+			if(currentNode.variable){
+				this._model.student.setVariable(newNodeID, currentNode.variable);
+				this._model.student.setStatus(newNodeID, "variable" , {"disabled":true,"status":"correct"});
+			}
+
 			if(currentNode.equation){
-				var inputs = [];
-				var isExpressionValid = true;
-				var equation = currentNode.equation;
-				array.forEach(currentNode.inputs, lang.hitch(this, function(input){
-					 var studentNodeID = this._model.student.getNodeIDFor(input.ID);
-					 if(studentNodeID){
-						inputs.push({ "ID": studentNodeID});
-						var regexp = "(" +input.ID +")([^0-9]?)";
-						var re = new RegExp(regexp);
-						equation = equation.replace(re, studentNodeID+"$2");
-					}
-					else{
-						isExpressionValid = false;
-					}
-				}));
-				if(isExpressionValid){
-					this._model.student.setInputs(inputs, newNodeID);
-					this._model.student.setEquation(newNodeID, equation);
-					this.givenEquationEntered = true;
+				var params = {
+					subModel: this._model.authored,
+					equation: currentNode.equation
+				};
+				var convert = equation.convert(params);
+				params = {
+					subModel: this._model.student,
+					equation: convert.equation,
+					nameToId: true,
+					autoCreateNodes: false
+				};
+				convert = equation.convert(params);
+				if(convert.success){
+					var links = [];
+					this._model.student.setLinks(convert.connections, newNodeID);
+					this._model.student.setEquation(newNodeID, convert.equation);
 					this._model.student.setStatus(newNodeID, "equation" , {"disabled":true,"status":"correct"});
-				}
-				else{
+				}else{
 					this._model.student.setInputs([], newNodeID);
 					this._model.student.setEquation(newNodeID, "");
-					this.errorStatus.push({"id": nodeid, "isExpressionCleared":true});
-					this._model.student.setStatus(newNodeID, "equation" , {"disabled":false,"status":"incorrect"});
+					//this.errorStatus.push({"id": nodeid, "isExpressionCleared":true});
+					//this._model.student.setStatus(newNodeID, "equation" , {"disabled":false,"status":"incorrect"});
```

Comment:
the conversion fails here because the equation has nodes that are not part of the model given to the student. So the equation is removed and I dont want it to show red initially so I removed it. Now it shows empty without the red color.